File: backend-java/src/main/java/septier/test/backend/models/AddManyRequest.java
package septier.test.backend.models;

import com.fasterxml.jackson.annotation.JsonProperty;

@SuppressWarnings({"WeakerAccess", "unused"})
public class AddManyRequest {
    public String DftrxId;
    public Config Config;
    public State State;
    public GpsExt GpsExt;
    public Compass Compass;
    public Integer Timestamp;
    public Point[] Points;

    public static class Config {
        public Integer RxGain;
        public Integer Network;
        @JsonProperty("2g_params")
        public CellNetworkParams Params2g;
        @JsonProperty("3g_params")
        public CellNetworkParams Params3g;
        @JsonProperty("4g_params")
        public CellNetworkParams Params4g;
        public Double Alpha;
        public String Slot;
        public Integer Celltrack;
        public Integer Watcher;
        public Integer Antenna;
        public Integer GpsSrc;
        public Integer Version;
        public Integer App;
        public Integer GsmMode;
        public String Url;
        public Double Sound;

        public static class CellNetworkParams {
            public Integer Arfcn;
            public Integer Ulsc;
            public Integer Dlsc;
        }
    }

    public static class State {
        public Integer DlMode;
        public String Tune;
        public Integer SignalFound;
        public Integer Hsdpa;
        public String Standard;
        public Integer Usrp;
        public Integer Capabilities;
        public Integer Expiration;
        public Integer Compression;
        public Integer Error;
        public Integer Remote;
    }

    public static class GpsExt {
        public Integer Status;
        public Double Lat;
        public Double Lon;
        public Double Alt;
        public Integer TS;
    }

    public static class Compass {
        public Double Hdg;
        public String FitErr;
    }

    public static class Point {
        public Integer Id;
        public Integer SettingsVersion;
        public Integer Channel;
        public Double SNR;
        public Angle[] Angles;
        public Double RSSI;
        public Double Angle2a;
        public Integer Antenna2a;
        public Integer Compression;
        public Integer Antenna;
        public Integer TSi;
        public Double TSf;
        public Long[] Clocks;

        public static class Angle {
            public Integer TSi;
            public Double TSf;
            public Double RSSI_m;
            public Double RSSI_s;
            public Double SNR_m;
            public Double SNR_s;
            public Integer Master;
            public Double An;
            public Double Phase;
            public Integer Visible;
            public Double Int_m;
            public Double Int_s;
            public Double Int;
            public Integer Ant;
            public Integer RxGain;
            public Integer V;
        }
    }
}

File: backend-go/server/models/add_many_request.go
package models

type AddManyRequest struct {
	DftrxId   string
	Config    Config
	State     State
	GpsExt    GpsExt
	Compass   Compass
	Timestamp int
	Points    []Point
}

type Config struct {
	RxGain    int
	Network   int
	Params_2g CellNetworkParams `json:"2g_params"`
	Params_3g CellNetworkParams `json:"3g_params"`
	Params_4g CellNetworkParams `json:"4g_params"`
	Alpha     float64
	Slot      string
	Celltrack int
	Watcher   int
	Antenna   int
	GpsSrc    int
	Version   int
	App       int
	GsmMode   int
	Url       string
	Sound     float64
}

type CellNetworkParams struct {
	Arfcn int
	Ulsc  int
	Dlsc  int
}

type State struct {
	DlMode       int
	Tune         string
	SignalFound  int
	Hsdpa        int
	Standard     string
	Usrp         int
	Capabilities int
	Expiration   int
	Compression  int
	Error        int
	Remote       int
}

type GpsExt struct {
	Status int
	Lat    float64
	Lon    float64
	Alt    float64
	TS     int
}

type Compass struct {
	Hdg    float64
	FitErr string
}

type Point struct {
	Id              int
	SettingsVersion int
	Channel         int
	SNR             float64
	Angles          []Angle
	RSSI            float64
	Angle2a         float64
	Antenna2a       int
	Compression     int
	Antenna         int
	TSi             int
	TSf             float64
	Clocks          []int64
}

type Angle struct {
	TSi     int
	TSf     float64
	RSSI_m  float64
	RSSI_s  float64
	SNR_m   float64
	SNR_s   float64
	Master  int
	An      float64
	Phase   float64
	Visible int
	Int_m   float64
	Int_s   float64
	Int     float64
	Ant     int
	RxGain  int
	V       int
}
